test/integration/scheduler/queue_test.go

```diff
@@ -849,27 +849,30 @@ func TestCoreResourceEnqueue(t *testing.T) {
 					}
 				}
 
-				// Wait for the tt.pods to be present in the scheduling queue.
+				// Wait for the tt.pods to be present in the scheduling active queue.
 				if err := wait.PollUntilContextTimeout(ctx, time.Millisecond*200, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
-					pendingPods, _ := testCtx.Scheduler.SchedulingQueue.PendingPods()
-					return len(pendingPods) == len(tt.pods), nil
+					return len(testCtx.Scheduler.SchedulingQueue.PodsInActiveQ()) == len(tt.pods), nil
 				}); err != nil {
 					t.Fatal(err)
 				}
 
 				t.Log("Confirmed Pods in the scheduling queue, starting to schedule them")
 
-				// Pop all pods out. They should be unschedulable.
+				// Pop all pods out. They should become unschedulable.
 				for i := 0; i < len(tt.pods); i++ {
 					testCtx.Scheduler.ScheduleOne(testCtx.Ctx)
 				}
-				// Wait for the tt.pods to be still present in the scheduling queue.
+				// Wait for the tt.pods to be still present in the scheduling (unschedulable) queue.
 				if err := wait.PollUntilContextTimeout(ctx, time.Millisecond*200, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
 					pendingPods, _ := testCtx.Scheduler.SchedulingQueue.PendingPods()
```

Comment:
```suggestion
					activePods := testCtx.Scheduler.SchedulingQueue.PodsInActiveQ()
					pendingPods, _ := testCtx.Scheduler.SchedulingQueue.PendingPods()
					return len(pendingPods)== len(tt.pods) && len(activePods)==0, nil
```
Why not just checking here instead of checking in L872.

Reply:
Well, in case someone puts schedulable nodes, the error message will be more clear and the test will fail sooner (without waiting for the timeout)

Reply:
> and the test will fail sooner

OK, makes sense.

We can do this though:
```
				// Wait for the tt.pods to be still present in the scheduling queue.
				if err := wait.PollUntilContextTimeout(ctx, time.Millisecond*200, wait.ForeverTestTimeout, false, func(ctx context.Context) (bool, error) {
					activePodsCount := len(testCtx.Scheduler.SchedulingQueue.PodsInActiveQ())
					if activePodsCount > 0 {
						return false, fmt.Errorf("Active queue was expected to be empty, but found %v Pods", activePodsCount)
					}

					pendingPods, _ := testCtx.Scheduler.SchedulingQueue.PendingPods()
					return len(pendingPods) == len(tt.pods), nil
				}); err != nil {
					t.Fatal(err)
				}
```

This way, it can fail even faster without waiting for the unscheduledPods.

Reply:
Potentially yes, but the time difference would be unnoticeable. On the other hand, in each run we could inconsistently report different number of violating pods, which could be misleading.

Reply:
1. We ensured that there are currently `len(tt.pods)` pods being scheduled in L855.
2. Suppose a pod mistakenly enters the activeQ now. While the other pods enters the unscheduledPod correctly.

Now, in current implementation, we will  never hit L875, because we `wait.ForeverTestTimeout`  in L867.
However, if we also check activeQ in `wait.ForeverTestTimeout`, we will fail the test immediately.

> Potentially yes, but the time difference would be unnoticeable. 

So I think the time difference is noticeable.

> On the other hand, in each run we could inconsistently report different number of violating pods, which could be misleading.

This should be fine. The test shouldn't be flaky at the first. It doesn't matter if the number is wrong, developer should test&fix it.  Integration test can be debugged locally, so it shouldn't be hard.

Reply:
FYI: we have a very low tolerance for flakiness in SIG Scheduling. So we don't need to optimize for consistency when the test flakes.